runtime/src/serde_utils.rs

```diff
@@ -1,64 +1,670 @@
+    fn into(self) -> AppendVec {
+        AppendVec::new_empty_map(self.current_len)
+    }
+}
+
+// Serialization of AppendVec V1_1_0 requires serialization of u64 to
+// eight byte vector which is then itself serialized to the stream
+impl Serialize for SerializableAppendVecV1_1_0 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        const LEN: usize = std::mem::size_of::<usize>();
+        let mut buf = [0u8; LEN];
+        serialize_into(Cursor::new(&mut buf[..]), &(self.current_len as u64))
+            .map_err(serde::ser::Error::custom)?;
+        serializer.serialize_bytes(&buf)
+    }
+}
+
+// Deserialization of AppendVec V1_1_0 requires deserialization
+// of eight byte vector from which u64 is then deserialized
+impl<'de> Deserialize<'de> for SerializableAppendVecV1_1_0 {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        use serde::de::Error;
+        struct SerializableAppendVecV1_1_0Visitor;
+        impl<'a> Visitor<'a> for SerializableAppendVecV1_1_0Visitor {
+            type Value = SerializableAppendVecV1_1_0;
+            fn expecting(&self, formatter: &mut Formatter) -> FormatResult {
+                formatter.write_str("Expecting SerializableAppendVecV1_1_0")
+            }
+            fn visit_bytes<E>(self, data: &[u8]) -> std::result::Result<Self::Value, E>
+            where
+                E: Error,
+            {
+                const LEN: u64 = std::mem::size_of::<usize>() as u64;
+                let mut rd = Cursor::new(&data[..]);
+                let current_len: usize = deserialize_from(&mut rd).map_err(Error::custom)?;
+                if rd.position() != LEN {
+                    Err(Error::custom(
+                        "SerializableAppendVecV1_1_0: unexpected length",
+                    ))
+                } else {
+                    Ok(SerializableAppendVecV1_1_0 { current_len })
+                }
+            }
+        }
+        deserializer.deserialize_bytes(SerializableAppendVecV1_1_0Visitor)
+    }
+}
+
+// Serializable version of AccountStorageEntry for snapshot format V1_1_1
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SerializableAccountStorageEntryV1_1_1 {
+    id: AppendVecId,
+    accounts_current_len: usize,
+}
+
+impl From<&AccountStorageEntry> for SerializableAccountStorageEntryV1_1_1 {
+    fn from(rhs: &AccountStorageEntry) -> Self {
+        Self {
+            id: rhs.id,
+            accounts_current_len: rhs.accounts.len(),
+        }
+    }
+}
+
+impl Into<AccountStorageEntry> for SerializableAccountStorageEntryV1_1_1 {
+    fn into(self) -> AccountStorageEntry {
+        AccountStorageEntry {
+            id: self.id,
+            accounts: AppendVec::new_empty_map(self.accounts_current_len),
+            ..AccountStorageEntry::default()
```

Comment:
I prefer to spell out all members here because when adding members in the future, we tend to forget to modify this function as lazy engineers. ;)

Reply:
The deserialisation code shouldn't be exposed to transient / non-persistent members of ASE.

Adding a dedicated ctor for ASE for the deserialistion code to use would be better.

Any naming preference?

Reply:
Yeah, that sounds better! then, let's follow the naming convention of `AppendVec::new_empty_map()`: so, **`AccountStorageEntry::new_empty(...)`**